presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/AgentFileUtils.java

```diff
@@ -130,4 +128,14 @@ public static void removePropertyFromFile(Path path, String property)
             properties.store(outputStream, null);
         }
     }
+
+    public static File downloadFile(URL url)
+            throws IOException
```

Comment:
This should take a URI, not a URL (though in the function, you'll want to convert it to a URL with `URI#toURL()`). 

This could be named `downloadTempFile`, but `downloadFile` is about as good.